### DISCUSSION TITLE
two question about spleen_segmentation_3d tutorial

### DISCUSSION THREAD

**Discussion by hz1z** [Project-MONAI/MONAI#7960] [General]
in you tutorial
[spleen_segmentation_3d](https://colab.research.google.com/github/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb#scrollTo=AxduhgY5cOhm)
# question1
```python
val_outputs = [post_pred(i) for i in decollate_batch(val_outputs)]
val_labels = [post_label(i) for i in decollate_batch(val_labels)]
```
Isn’t this redundant? This code decollates the data from the batch dimension and then wraps it into a list, effectively doing nothing, right? Why not directly use post_pred(val_outputs) and post_label(val_labels)? Does it mean that the AsDiscrete transform has to remove the data from the batch dimension to work?
# question2
```python
train_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        EnsureChannelFirstd(keys=["image", "label"]),
        ScaleIntensityRanged(
            keys=["image"],
            a_min=-57,
            a_max=164,
            b_min=0.0,
            b_max=1.0,
            clip=True,
        ),
        CropForegroundd(keys=["image", "label"], source_key="image"),
        Orientationd(keys=["image", "label"], axcodes="RAS"),
        Spacingd(keys=["image", "label"], pixdim=(1.5, 1.5, 2.0), mode=("bilinear", "nearest")),
        RandCropByPosNegLabeld(
            keys=["image", "label"],
            label_key="label",
            spatial_size=(96, 96, 96),
            pos=1,
            neg=1,
            num_samples=4,
            image_key="image",
            image_threshold=0,
        ),
# for inference
roi_size = (160, 160, 160)
sw_batch_size = 4
```
why roi_size = (160, 160, 160)?
 for training,you randomly crop the data used  to a size of (96, 96, 96)? Why are you using (160, 160, 160) for inference instead?

**Comment by KumoLiu** [comment]
Hi @hz1z, thanks for your interest here.

> Isn’t this redundant? This code decollates the data from the batch dimension and then wraps it into a list, effectively doing nothing, right? Why not directly use post_pred(val_outputs) and post_label(val_labels)? Does it mean that the AsDiscrete transform has to remove the data from the batch dimension to work?

In MONAI, all transforms assume input data shape as [channel x spatial_dims]. 
https://github.com/Project-MONAI/MONAI/wiki/MONAI-Data-Shape-Assumption

> for training,you randomly crop the data used to a size of (96, 96, 96)? Why are you using (160, 160, 160) for inference instead?

Using random clipping to a smaller size (96x96x96) during the training phase and a larger size (160x160x160) during the inference phase is a common strategy that aims to balance the generalization ability of the model and the accuracy in real-world applications. With this approach, the model is able to learn richer features during training and make full use of them during inference to improve performance.

convert to discussion more like a usage issue, feel free to create another if meet any other issue. Thanks!

**Comment by hz1z** [comment]
thank you ！ @KumoLiu 
Regarding the first issue, I have already understood why it is necessary to remove the data from the batch dimension and then repackage it into the batch dimension after referring to the [wiki](https://github.com/Project-MONAI/MONAI/wiki/MONAI-Data-Shape-Assumption) you recommended. 

As for the second issue, I seem to have grasped it; it might be related to the local characteristics of CNNs. It is possible to train on smaller patches and infer on larger patches without affecting the network accuracy.
